pkg/pulumiyaml/run.go

```diff
@@ -343,6 +378,14 @@ func (r *Runner) setDefaultProviders() {
 	contract.IgnoreError(diags)
 }
 
+// Set the runners package descriptors from the templates package decls.
+func (r *Runner) setPackageDesciptors() error {
+	// Register package refs for all packages we know upfront
+	packageDescriptors, err := packages.ToPackageDescriptors(r.t.Packages)
+	r.packageDescriptors = packageDescriptors
+	return err
```

Comment:
Should we always set even if there is an error?

Reply:
Probably not